packages/volto/src/middleware/api.js

```diff
@@ -188,7 +187,15 @@ const apiMiddlewareFactory =
                     ),
                   },
                 ).then((reqres) => {
-                  dispatch(updateUploadedFiles(uploadedFiles++));
+                  if (item.op === 'post') {
+                    let uploadedFiles =
+                      getState().content.subrequests?.['batch-upload']
+                        ?.uploadedFiles;
+
+                    dispatch(
+                      updateUploadedFiles(++uploadedFiles, 'batch-upload'),
```

Comment:
I was thinking more in something more simple like 

```js
if (action.subrequest === 'batched-upload') {
    dispatch(updateUploadedFiles(++uploadedFiles))
}
```
and that's it. Without anything that you did in the last commit. Is it possible?

Reply:
oh! that's what you meant.
Lol. I unnecessarily came up with something too complicated. 

I Tried out the above example. It works fine. 

[Screencast from 07-04-24 02:21:48 PM IST.webm](https://github.com/plone/volto/assets/89346667/7cd308e9-5e84-4907-9522-1d2ed248aff9)